example/src/main/java/com/appunite/rx/example/MainActivity.java

```diff
@@ -48,52 +50,23 @@
     @InjectView(R.id.main_activity_fab)
     FloatingActionButton fab;
 
-    public static class AdapterItemManager implements ViewHolderManager {
-
-        @Override
-        public boolean matches(@Nonnull BaseAdapterItem baseAdapterItem) {
-            return baseAdapterItem instanceof MainPresenter.AdapterItem;
-        }
+    private class MainViewHolder extends SubscriptionViewHolder<MainPresenter.AdapterItem> {
```

Comment:
this solution has some disadvantages from previous one. 
`AdapterItemManager extends ViewHolderManager` is static class it can be injected via dagger and `MainViewHolder` is not static so it can used injected resources from `AdapterItemManager`.

so implementation of `ViewHolderManager` are actually some kind of factory for `BaseViewHolder`